app/controllers/application_controller/filter.rb

```diff
@@ -866,9 +866,9 @@ def adv_search_clear
             adv_search_build(vm_model_from_active_tree(x_active_tree))
           end
           session[:edit] = @edit              # Set because next method will restore @edit from session
-          listnav_search_selected(search_id)  # Clear or set the adv search filter
-          self.x_node = "root"
         end
+        listnav_search_selected(search_id)  # Clear or set the adv search filter
+        self.x_node = "root"
```

Comment:
Only moved these 2 lines out of the conditional. There doesn't seem to be a reason for them not to apply to all views with filters/trees.